src/frontend/components/UploadForm/index.tsx

```diff
@@ -0,0 +1,217 @@
+    description: 'Title for the video upload form',
+    id: 'components.UploadForm.title-videos',
+  },
+  [modelName.THUMBNAIL]: {
+    defaultMessage: 'Upload a new thumbnail',
+    description: 'Title for the thumbnail upload form',
+    id: 'components.UploadForm.title-thumbnail',
+  },
+});
+
+const timedtexttrackTitleMessages = defineMessages({
+  [timedTextMode.CLOSED_CAPTIONING]: {
+    defaultMessage: 'Upload a new closed captions file',
+    description: 'Title for the timed text file upload form',
+    id: 'components.UploadForm.title-timedtexttracks-cc',
+  },
+  [timedTextMode.SUBTITLE]: {
+    defaultMessage: 'Upload a new subtitles file',
+    description: 'Title for the timed text file upload form',
+    id: 'components.UploadForm.title-timedtexttracks-st',
+  },
+  [timedTextMode.TRANSCRIPT]: {
+    defaultMessage: 'Upload a new transcript file',
+    description: 'Title for the timed text file upload form',
+    id: 'components.UploadForm.title-timedtexttracks-ts',
+  },
+});
+
+const UploadFormContainer = styled(LayoutMainArea)`
+  display: flex;
+  flex-direction: column;
+  align-items: stretch;
+`;
+
+const IframeHeadingWithLayout = styled(IframeHeading)`
+  flex-grow: 0;
+  margin: 0;
+  text-align: center;
+`;
+
+const UploadFieldContainer = styled.div`
+  flex-grow: 1;
+  display: flex;
+`;
+
+const UploadFormBack = styled.div`
+  line-height: 2rem;
+  padding: 0.5rem 1rem;
+`;
+
+/** Props shape for the BaseUploadForm component. */
+interface BaseUploadFormProps {
+  jwt: string;
+  notifyObjectUploadProgress: (
+    progress: UploadProgressNotification<UploadableObject>['progress'],
+  ) => void;
+  object: Maybe<UploadableObject>;
+  objectType: modelName;
+  updateObject: (object: UploadableObject) => void;
+}
+
+export type Status = Maybe<
+  'not_found_error' | 'policy_error' | 'uploading' | 'success'
+>;
+
+/**
+ * Component. Displays a form with a dropzone and a file upload button to add a file to the
+ * object passed in the props.
+ * @param jwt The token that will be used to fetch the object record from the server to update it.
+ * @param notifyObjectUploadProgress Action creator that takes the object's upload progress to update it in the store.
+ * @param object The object for which we want to add a file.
+ * @param objectType The model name for the object.
+ * @param updateObject Action creator that takes an object to update it in the store.
+ */
+const BaseUploadForm = ({
+  jwt,
+  notifyObjectUploadProgress,
+  object,
+  objectType,
+  updateObject,
+}: BaseUploadFormProps) => {
+  const [status, setStatus] = useState(undefined as Status);
+
+  const beforeUnload = (event: BeforeUnloadEvent) => {
+    if (status === 'uploading') {
+      event.preventDefault();
+      event.returnValue = '';
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener('beforeunload', beforeUnload);
+
+    return () => window.removeEventListener('beforeunload', beforeUnload);
+  }, [status]);
```

Comment:
```suggestion
  }, []);
```

Why is status a dependency here?

Reply:
I only want to update the `useEffect` when the status change, I use it in my function.

Reply:
I don't understand this.

It was previously in `componentDidMount`, so only ran once when the component mounted. Now you're adding a new `'beforeunload'` listener every time the `status` changes.

What am I missing?